lib/cpp/src/thrift/server/TThreadedServer.cpp

```diff
@@ -90,31 +95,71 @@ TThreadedServer::~TThreadedServer() {
 }
 
 void TThreadedServer::serve() {
+  threadFactory_->setDetached(false);
```

Comment:
This is a functional change that is subtle and may break existing users of TThreadedServer. I don't have a better solution at the moment, but I do want to call this out to see if others have a better solution.